docs/source/opkg-keyrings/opkg-keyrings_index.rst

```diff
@@ -0,0 +1,157 @@
+
+The opkg-keyrings package is updated periodically to add new keys or remove
+expired keys. It is important to keep the keyrings up-to-date to ensure that
+packages can be verified and installed correctly.
+
+This document provides instructions on how to upgrade the opkg-keyrings
+package on NI Linux Real-Time systems.
+
+Purpose of the opkg-keyrings package
+-------------------------------------
+#. Store Trusted Signing Keys:
+
+    * The package includes the public keys required to verify .ipk package signatures.
+    * These keys are placed in specific directories, such as /usr/share/opkg/keyrings/.
+
+#. Ensure Package Integrity:
+
+    * During installation, opkg checks package signatures using these keys.
+    * This prevents unauthorized or tampered packages from being installed.
+
+#. Key Management:
+
+    * The opkg-keyrings package is updated when signing keys are rotated (e.g., when
+        old keys are retired, and new ones are issued).
+    * Upgrading this package ensures the keyring remains current and secure.
+
+How opkg signing works and why we do it
+=======================================
+opkg signing is a mechanism to ensure the authenticity and integrity of .ipk
+packages being installed on a system. It involves cryptographic signatures that
+verify packages come from a trusted source and have not been tampered with. 
+
+How does it work?
+-----------------
+When a package is built, it is signed with a private key. The corresponding
+public key is included in the opkg-keyrings package. When the package is
+downloaded, opkg checks the package signature against the public key to
+verify the package's authenticity and integrity.
+
+Steps involved:
+----------------
+
+#. Generate a Signing Key:
+
+    * A pair of private and public keys is created using a tool like gpg.
+    * The private key is used to sign the package index or individual packages
+        (package index, the *Packages* and *Packages.gz* files in the case of NILRT).
+    * The public key is distributed to the target systems.
+
+#. Sign the Package Index:
+
+    * The package index (*Packages* file) lists all available packages and their
+        metadata. This file is signed with the private key.
+    * A detached signature (*Packages.asc*) is generated and placed alongside the
+        *Packages* file.
+
+#. Verify Signature During Package Installation:
+
+    * When opkg installs a package, it verifies the signature of the *Packages* file
+        or the package itself using the public key.
+    * The public key is preloaded or added to the system's trusted keyring, such as
+        opkg-key (see :ref:`How keys are added to the opkg-keyring` for more details
+        on how this is accomplished).
+
+#. Trust Management:
+
+    * Trusted keys are stored in a keyring directory (e.g., /usr/share/opkg/keyrings/).
+    * During package installation, opkg ensures that the signature matches one of the
```

Comment:
Wording makes it sound like the signatures are string compared to key but they're cryptographically verified.
And is the feed's signature verified during package installation or just during `opkg update` operation? I know that package's md5/sha256sum are verified during package installation but that's not the same as "signatures matching trusted keys".

Reply:
I have rephrased this.